### PR TITLE
Initial sql/api work for 3096

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-3096](https://issues.redhat.com/browse/COST-3096)

## Description

This change will add a new param to the API's for bucketing default OpenShift platform cost/usage into a new platform category, when doing a group_by by project. New param is `category` current options are `*` or `platform`

This also includes 
- A classification annotation so you can tell the difference between a project and a category.
- A default_project annotation to quickly tell if a project is a Default project (openshift platform/default openshift cluster owned project)

## Testing

1. Checkout Branch
2. Restart Koku
3. Ingest OCP data (ideally with correlated cloud types to check all API's) Note for platform costs this needs to include projects starting with `openshift-` and/or `kube-` (These would be classed as openshift default projects)
4. Navigate to API's with the following params: 
    1. http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&category=* 
        - Result is a platform category with all usage/cost from `openshift-` and `kube-` projects condensed into a single line item
        - All remaining projects returned like normal
        - Plus a classification annotation to distinguish between a _real_ project and a category (group of projects) 
    2. http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&category=platform 
        - Exactly the same as above
    3. http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=* 
        - Result is projects returned like normal but with an additional `default_project` annotation with true/false values to distinguish between a user created project and a openshift/cluster created project (default project)
    4. http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&category=test
        - Result should be validation error test is not a valid category. 

## Notes

Some examples

?group_by[project]=*&category=platform

      "project": "platform",
      "classification": "category",

      "project": "my-project",
      "classification": "project",

?group_by[project]=*

      "project": "openshift-something",
      "default_project": "True",

      "project": "my-project",
      "default_project": "False",

?group_by[project]=*&category=test

"Unsupported category: test"
...

### CODE REVIEW

**Comment by lcouzens** [comment]
Amazing only one Smoke failure and it's expected from main test_api_ocp_on_aws_source_raw_calc

File: koku/api/report/queries.py [hunk 1]
```diff
@@ -1017,6 +1040,15 @@ def _aggregate_ranks_over_limit(self, data_frame, group_by):
         other_str = "Others" if other_count > 1 else "Other"
         for group in group_by:
             others_data_frame[group] = other_str
+            if (
+                "project" in self.parameters.parameters.get("group_by", {})
+                or "and:project" in self.parameters.parameters.get("group_by", {})
+                or "or:project" in self.parameters.parameters.get("group_by", {})
+            ):
```
Comment (maskarb) [inline koku/api/report/queries.py:1047]:
```suggestion
            if is_grouped_by_project(self.parameters):
```

**Review by maskarb — CHANGES_REQUESTED** [review]

**Comment by maskarb** [comment]
Is `category` only valid when `group_by[project]` is used? Should it be a validation error if `category` is used when we are not grouping by project?

File: koku/api/query_params.py [hunk 2]
```diff
@@ -462,6 +463,23 @@ def delta(self):
         """Return delta property."""
         return self.get("delta")
 
+    @property
+    def category(self):
```
[the remark this answers is not in the document]
Comment (lcouzens) [reply koku/api/query_params.py:467]: Thanks Michael, I figured this was wrong and you would set me straight. :bow:

**Review by maskarb — DISMISSED** [review]
lgtm!

**Review by esebesto — APPROVED** [review]

**Pull request merged**